packages/protocol-http/src/extensions/http.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpHandler } from "../httpHandler";
+import { HttpHandlerExtensionConfiguration } from "./httpExtensionConfiguration";
+
+/**
+ * @internal
+ */
+export const getHttpHandlerConfiguration = <HandlerConfig extends object = Record<string, unknown>>(
+  runtimeConfig: Partial<{ httpHandler: HttpHandler<HandlerConfig> }>
+) => {
+  let httpHandler = runtimeConfig.httpHandler!;
+  const httpHandlerConfigs: HandlerConfig = httpHandler.httpHandlerConfigs();
+  return {
+    setHttpHandler(handler: HttpHandler<HandlerConfig>): void {
+      httpHandler = handler;
+    },
+    httpHandler(): HttpHandler<HandlerConfig> {
+      return httpHandler;
+    },
+    updateHttpClientConfig(key: keyof HandlerConfig, value: HandlerConfig[typeof key]): void {
+      httpHandlerConfigs[key] = value;
+    },
+    httpHandlerConfigs(): HandlerConfig {
+      return httpHandlerConfigs;
+    },
+  };
+};
+
+/**
+ * @internal
+ */
+export const resolveHttpHandlerRuntimeConfig = <HandlerConfig extends object = Record<string, unknown>>(
+  httpHandlerExtensionConfiguration: HttpHandlerExtensionConfiguration<HandlerConfig>
+) => {
+  const requestHandler = httpHandlerExtensionConfiguration.httpHandler();
+  const handlerConfig = httpHandlerExtensionConfiguration.httpHandlerConfigs();
+  for (const key in handlerConfig) {
+    requestHandler.updateHttpClientConfig(key, handlerConfig[key]);
+  }
```

Comment:
why is this code in this position?

it looks like it just tells a request handler to set its config to values provided by its own config, redundant?